### PR TITLE
fix(removed): Make sure removed entities do not appear on recommendations

### PR DESCRIPTION
**Pull request by dexter-mh-lee**
Also apply removed filter on the metadata analytics charts. 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-io/src/main/java/com/linkedin/metadata/recommendation/candidatesource/MostPopularSource.java [hunk 1]
```diff
@@ -106,7 +110,7 @@ private SearchRequest buildSearchRequest(@Nonnull Urn userUrn) {
     // Find the entities with the most views
     AggregationBuilder aggregation = AggregationBuilders.terms(ENTITY_AGG_NAME)
         .field(DataHubUsageEventConstants.ENTITY_URN + ESUtils.KEYWORD_SUFFIX)
-        .size(MAX_CONTENT);
+        .size(MAX_CONTENT * 2);
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/recommendation/candidatesource/MostPopularSource.java:113]: Curious - why this change necessary?

Comment (dexter-mh-lee) [reply metadata-io/src/main/java/com/linkedin/metadata/recommendation/candidatesource/MostPopularSource.java:113]: so that after filtering we still have the num left

**Review by shirshanka — APPROVED** [review]
